capsules/extra/src/hmac.rs

```diff
@@ -384,28 +374,30 @@ impl<
 
 impl<
         'a,
-        H: digest::Digest<'a, L> + digest::HmacSha256 + digest::HmacSha384 + digest::HmacSha512,
-        const L: usize,
-    > digest::ClientHash<L> for HmacDriver<'a, H, L>
+        H: digest::Digest<'a, D> + digest::HmacSha<S>,
+        D: digest::DigestAlgorithm,
+        S: digest::ShaAlgorithm,
+    > digest::ClientHash<D> for HmacDriver<'a, H, D, S>
 {
-    fn hash_done(&self, result: Result<(), ErrorCode>, digest: &'static mut [u8; L]) {
+    fn hash_done(&self, result: Result<(), ErrorCode>, digest: &'static mut D::Digest) {
         self.processid.map(|id| {
             self.apps
                 .enter(id, |_, kernel_data| {
                     self.hmac.clear_data();
 
-                    let pointer = digest[0] as *mut u8;
+                    let pointer = digest.as_ref()[0] as *mut u8;
```

Comment:
What is this `pointer` intended for? It looks like it is supposed to return the address of the kernel buffer to userspace? But in practice it just creates a pointer from the first byte of the sha256 hash, and then casts it back to a usize before passing to userspace? The effect is passing the first byte in the upcall, but I'm not sure why?

Reply:
I don't think this is used and can just be removed